source/bank.py

```diff
@@ -217,13 +263,69 @@ def redeem_promissory_note(self, note):
                 check.owner_public_key)
 
             if buyer_device_data.is_unspent(check):
-                buyer_device_data.spend_check(check)
+                # This case can only occur if the buyer didn't already hand the note to their bank before.
+                if affects_cap and is_claimable:
+                    buyer_device_data.spend_check(check, amount)
+                else:
+                    buyer_device_data.spend_check(check)
+            elif note.draft in buyer_device_data.awaiting_claim:
+                # This case occurs when the note was handed in before by the buyer, and the unspent checks have already been cleared.
+                # If the note expired and the transaction date falls in the current month, restore the note's value to the spending
+                # cap for this month.
+                if not is_claimable and affects_cap:
+                    buyer_device_data.cap += amount
+            elif not is_claimable:
+                # This case occurs when the note was handed in before by the buyer and the unspent checks have already been cleared,
+                # but has already been removed from the 'awaiting claim' set again by the bank itself because it expired.
+                pass
+            elif check.unredeemable:
+                # This case occurs when the note is still claimable but somehow contains an unredeemable check
+                raise FraudException(
+                    'Oh lawd %s used expired checks for the transaction!' % buyer_account.owner)
             else:
                 raise FraudException(
-                    'Oh lawd %s is double-spending!' % buyer_account.owner)
+                    'Oh lawd %s is double-spending or %s is double-redeeming!' % buyer_account.owner, seller_account.owner)
```

Comment:
I think double-redeeming should actually be fine. There's no way sellers can hope to cheat the bank by double-redeeming, right (the bank will always check if the note can be redeemed before actually redeeming). We might want to just treat this as an honest mistake and make it a no-op.